### ISSUE TITLE
add countDistinct on GroupBy

### ISSUE THREAD

**Issue by devcrocod** [Kotlin/dataframe#533]
I would like to have a function `countDistinct()` on grouped dataframe for counting distinct values by group

```kotlin
df.groupBy { column_0 }.countDistinct()
```

```kotlin
df.groupBy { column_0 }.countDistinct { column_1 }
```

**Comment by Jolanrensen** [comment]
but then it should probably be named `countDistinctValues()` right?

**Comment by devcrocod** [comment]
No, it's similar https://kotlin.github.io/dataframe/countdistinct.html but on grouped df. I wouldn't like the naming to become more complicated

**Comment by devcrocod** [comment]
this is similar to other functions that apply to both a dataframe and a grouped dataframe: `count`, `sum`, `mean` ...
